ogr/ogrsf_frmts/hana/ogrhanatablelayer.cpp

```diff
@@ -0,0 +1,1762 @@
+                continue;
+        }
+        values.push_back(
+            QuotedIdentifier(clmDesc.name) + " = "
+            + GetParameterValue(
+                clmDesc.type, clmDesc.typeName, clmDesc.isArray));
+        if (clmDesc.isArray)
+            hasArray = true;
+    }
+
+    for (const GeometryColumnDescription& geomClmDesc : geomColumns_)
+    {
+        values.push_back(
+            QuotedIdentifier(geomClmDesc.name) + " = " + "ST_GeomFromWKB(?, "
+            + std::to_string(geomClmDesc.srid) + ")");
+    }
+
+    if (hasArray && !parseFunctionsChecked_)
+    {
+        // Create helper functions if needed.
+        if (!dataSource_->ParseArrayFunctionsExist(schemaName_.c_str()))
+            dataSource_->CreateParseArrayFunctions(schemaName_.c_str());
+        parseFunctionsChecked_ = true;
+    }
+
+    const CPLString sql = StringFormat(
+        "UPDATE %s SET %s WHERE %s = ?",
+        GetFullTableNameQuoted(schemaName_, tableName_).c_str(),
+        JoinStrings(values, ", ").c_str(),
+        QuotedIdentifier(GetFIDColumn()).c_str());
+
+    return dataSource_->PrepareStatement(sql.c_str());
+}
+
+/* -------------------------------------------------------------------- */
+/*                     ResetPreparedStatements()                        */
+/* -------------------------------------------------------------------- */
+
+void OGRHanaTableLayer::ResetPreparedStatements()
+{
+    if (!currentIdentityValueStmt_.isNull())
+        currentIdentityValueStmt_ = nullptr;
+    if (!insertFeatureStmtWithFID_.isNull())
+        insertFeatureStmtWithFID_ = nullptr;
+    if (!insertFeatureStmtWithoutFID_.isNull())
+        insertFeatureStmtWithoutFID_ = nullptr;
+    if (!deleteFeatureStmt_.isNull())
+        deleteFeatureStmt_ = nullptr;
+    if (!updateFeatureStmt_.isNull())
+        updateFeatureStmt_ = nullptr;
+}
+
+/************************************************************************/
+/*                        SetStatementParameters()                      */
+/************************************************************************/
+
+OGRErr OGRHanaTableLayer::SetStatementParameters(
+    odbc::PreparedStatement& statement,
+    OGRFeature* feature,
+    bool newFeature,
+    bool withFID,
+    const char* functionName)
+{
+    OGRHanaFeatureReader featReader(*feature);
+
+    unsigned short paramIndex = 0;
+    int fieldIndex = -1;
+    for (const AttributeColumnDescription& clmDesc : attrColumns_)
+    {
+        if (clmDesc.isFeatureID)
+        {
+            if (!withFID && clmDesc.isAutoIncrement)
+                continue;
+
+            ++paramIndex;
+
+            switch (clmDesc.type)
+            {
+            case odbc::SQLDataTypes::Integer:
+                if (feature->GetFID() == OGRNullFID)
+                    statement.setInt(paramIndex, odbc::Int());
+                else
+                {
+                    if (!CanCastIntBigTo<std ::int32_t>(feature->GetFID()))
```

Comment:
```suggestion
                    if (!CanCastIntBigTo<std::int32_t>(feature->GetFID()))
```

Reply:
Reworked in c9e8b4bf2dcc5962db78e8063d5c01c8e152b931.